### PR TITLE
Add support for AnyObject in parameters

### PR DESCRIPTION
**Pull request by pietbrauer**
Following up from #11 and #10 I went ahead and implemented the serialisation of `[String: AnyObject]` for parameters in a URL request, as well as JSON.

/cc @1ec5

### CODE REVIEW

**Comment by 1ec5** [comment]
This looks good at a glance. But I wonder if it’d be cleaner to build the request URL using a NSURLComponents object (whose `queryItems` array holds NSURLQueryItem objects) instead of building the URL manually using pure Swift Standard Library types.

**Comment by pietbrauer** [comment]
Yes, I think this might be the best idea. Thanks for the feedback!

**Comment by pietbrauer** [comment]
@1ec5 Ready to review now. Don't know why travis got the wrong SHA checked out for the PullRequest.

File: RequestKit/Router.swift [hunk 1]
```diff
@@ -31,61 +31,73 @@ public protocol Router {
...
+            guard let value = parameters[key] else { continue }
+            switch value {
+            case let value as String:
+                if let escapedValue = value.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLQueryAllowedCharacterSet()) {
+                    components.append(NSURLQueryItem(name: key, value: escapedValue))
+                }
+            case let valueArray as [String]:
+                valueArray.forEach { item in
+                    if let escapedValue = item.stringByAddingPercentEncodingWithAllowedCharacters(NSCharacterSet.URLQueryAllowedCharacterSet()) {
+                        components.append(NSURLQueryItem(name: "\(key)[\(valueArray.indexOf(item)!)]", value: escapedValue))
```
Comment (1ec5) [inline RequestKit/Router.swift]: Instead of calling indexOf inside a forEach closure, use a simple for loop over the range `0..<valueArray.count`.

Comment (1ec5) [inline RequestKit/Router.swift]: If you prefer a functional approach, you could `enumerate` the array.

**Comment by 1ec5** [comment]
This looks great aside from the one comment. Thanks!

**Comment by pietbrauer** [comment]
@1ec5 Fixed it, thanks for reviewing!

**Pull request merged**